src/controllers/accounts.controller.js

```diff
@@ -47,24 +49,43 @@ exports.register = async (req, res, next) => {
     return res.status(422).json(responseMsg.validationError422(validation.errors));
   }
 
-  // check whether the email has been used
   const {email} = req.body;
-  const existed = await dynamoDb.getUser(email);
-  if (!existed.success) {
-    return res.status(500).json(existed);
-  } else if (existed.data.length > 0) {
-    return res.status(422).json(responseMsg.error(errorMsg.params.EMAIL,
-        errorMsg.messages.EMAIL_REGISTERED));
-  }
 
   // build item
   const item = {};
   for (const key in req.body) {
-    if (key === 'referral') continue; // TODO update when referral code has been defined
-    // hash the password
-    item[key] = req.body[key];
-    if (key === 'password') {
-      item[key] = bcrypt.hashSync(req.body[key], salt);
+    if (req.body.hasOwnProperty(key)) {
```

Comment:
eslint will complain because of Object.prototypes (which can lead to unexpected behavior, but in our case it should be okay).

https://stackoverflow.com/questions/4166551/javascript-jslint-error-the-body-of-a-for-in-should-be-wrapped-in-an-if-statem/4166584
https://eslint.org/docs/rules/no-prototype-builtins

Also was reallyyyyyy fun to debug this